openfe/setup/ligand_network_planning.py

```diff
@@ -82,67 +99,57 @@ def generate_radial_network(
     """
     if isinstance(mappers, AtomMapper):
         mappers = [mappers]
-    mappers = [_hasten_lomap(m, ligands) if isinstance(m, LomapAtomMapper)
-               else m for m in mappers]
+    mappers = [
+        _hasten_lomap(m, ligands) if isinstance(m, LomapAtomMapper) else m
+        for m in mappers
+    ]
 
     # handle central_ligand arg possibilities
     # after this, central_ligand is resolved to a SmallMoleculeComponent
     if isinstance(central_ligand, int):
         ligands = list(ligands)
         try:
             central_ligand = ligands[central_ligand]
+            ligands.remove(central_ligand)
         except IndexError:
-            raise ValueError(f"index '{central_ligand}' out of bounds, there are "
-                             f"{len(ligands)} ligands")
+            raise ValueError(
+                f"index '{central_ligand}' out of bounds, there are "
+                f"{len(ligands)} ligands"
+            )
     elif isinstance(central_ligand, str):
         ligands = list(ligands)
         possibles = [l for l in ligands if l.name == central_ligand]
         if not possibles:
-            raise ValueError(f"No ligand called '{central_ligand}' "
-                             f"available: {', '.join(l.name for l in ligands)}")
+            raise ValueError(
+                f"No ligand called '{central_ligand}' "
+                f"available: {', '.join(l.name for l in ligands)}"
+            )
         if len(possibles) > 1:
             raise ValueError(f"Multiple ligands called '{central_ligand}'")
         central_ligand = possibles[0]
+        ligands.remove(central_ligand)
```

Comment:
I think you can move this outside the `elif` (and remove the call on line 113), since you will be removing the central ligand in either case.

Reply:
and remove line 113

Reply:
I believe line 112 would throw an error if the `central _ligand` element is not in the list.

Reply:
True, but i think the if/elif branches do not cover the case if `central_ligand` is of type `Component`